flows/actions/send_msg.go

```diff
@@ -127,14 +128,16 @@ func (a *SendMsgAction) getTemplateMsg(run flows.Run, urn urns.URN, channelRef *
 		evaluatedVariables[i] = v
 	}
 
-	// cross-reference with asset to get variable types
+	// cross-reference with asset to get variable types and filter out invalid values
 	variables := make([]*flows.TemplatingVariable, len(translation.Variables()))
 	for i, v := range translation.Variables() {
+		// we pad out any missing variables with empty values
+		value := ""
 		if i < len(evaluatedVariables) {
-			variables[i] = &flows.TemplatingVariable{Type: v.Type(), Value: evaluatedVariables[i]}
-		} else {
-			variables[i] = &flows.TemplatingVariable{Type: v.Type(), Value: ""}
+			value = evaluatedVariables[i]
 		}
+
+		variables[i] = &flows.TemplatingVariable{Type: v.Type(), Value: value}
```

Comment:
I thought about having this ignore attachments that don't match the header type, e.g. ignore `video/mp4:http://...` if variable is type `image`... but the empty values will cause a send failure anyway and I think it will be more obvious to the user what the problem is if we just _try_ to send the wrong attachment type and let WhatsApp complain about it.

Reply:
I avoided the mismatch by using the attachment type in https://github.com/nyaruka/courier/pull/752/files should I keep using the parameter type instead?
Of course both approaches would fail to send if the parameter type is not the same as the one on the template

Reply:
I think your change is fine.. in the case that someone updates their template but doesn't update the flow using it, we'll send a param of the wrong type to whatsapp, and I assume they'll send back a fairly clear error message.